src/languages/java.ts

```diff
@@ -0,0 +1,76 @@
+import { patternFinder } from "../nodeFinders";
+import {
+  matcher,
+  createPatternMatchers,
+  argumentMatcher,
+  notSupported,
+} from "../nodeMatchers";
+import { selectWithLeadingDelimiter } from "../nodeSelectors";
+import { NodeMatcherAlternative, ScopeType } from "../Types";
+
+// Generated by the following command:
+// > curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-java/master/src/node-types.json | jq '[.[] | select(.type == "statement" or .type == "declaration") | .subtypes[].type]'
+const STATEMENT_TYPES = [
+  "annotation_type_declaration",
+  "class_declaration",
+  "enum_declaration",
+  "import_declaration",
+  "interface_declaration",
+  "module_declaration",
+  "package_declaration",
+  //   ";",
+  "assert_statement",
+  "block",
+  "break_statement",
+  "continue_statement",
+  "declaration",
+  "do_statement",
+  "enhanced_for_statement",
+  "expression_statement",
+  "for_statement",
+  "if_statement",
+  "labeled_statement",
+  "local_variable_declaration",
+  "return_statement",
+  "switch_expression",
+  "synchronized_statement",
+  "throw_statement",
+  "try_statement",
+  "try_with_resources_statement",
+  "while_statement",
+  "yield_statement",
+];
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  class: "class_declaration",
+  className: "class_declaration[name]",
+  namedFunction: ["method_declaration", "constructor_declaration"],
+  functionName: [
+    "method_declaration.identifier!",
+    "constructor_declaration.identifier!",
+  ],
+  ifStatement: "if_statement",
+  string: "string_literal",
+  name: ["*[declarator.name]", "*[name]", "formal_parameter.identifier!"],
```

Comment:
What about 
```suggestion
  name: ["*[declarator][name]", "*[name]", "formal_parameter.identifier!"],
```

I think reserving dot for child is clearer

Reply:
But it is a child. Is(name) the child node of the declarator node

Reply:
Right but "name" is the field name not the child type, as a dot usually indicates, no?

Reply:
In both cases not just mean a child to a parent. In one case its type in one the case it field names but both are parent child. If you feel strongly about it I can try to fix a regular expression for your format.

Reply:
So square brackets and dots always mean children, right? But square bracket indicates a field name, and dot indicates a type name. So I'd stick to that convention if that's the case

Reply:
Correct

Reply:
Okay get I'll try to get that work

Reply:
Fixed!

Reply:
Awesome! Looks like it simplified the code a bit too so that's always a good sign

Reply:
Yea if you don't count all the time it took to write all those regular expressions ;)